### ISSUE TITLE
Memory requirement detail during Training

### ISSUE THREAD

**Issue by Zhanpeng1202** [facebookresearch/vggt#189]
Hi,

Thanks for opening sources this great work.

We are implementing a finetune pipe for vggt.  But the memory usage go very high, I can only enable 1 batch for each 48G GPU. I think my current problems probably lie in the gradient checkpointing.

Would you like to explain these questions:

1. As stated in the issue #34 , FlashAttention v2 has been intergrated into Pytorch > 2.3 Therefore we do not need to include xformers in the training ?
2. As discussed in #31 , you enable BF16 as precission. But during training, should we use float16-true or 16-mixerd?

**Comment by jytime** [comment]
1. Right xformers should not be used.
2. Can you explain what is float16-true or 16-mixerd? It sounds like some flags not naturally supported by pytorch, probably from Lightning? We did not use it.

**Comment by Zhanpeng1202** [comment]
Thanks a lot for answering! 

Yes, it is an API in the Pytorch Lightning. I think 16-mixed set model weight as float 32 and set data as float 16 , while 16-true use float16 for both data and model.

Also for the head prediction, you disable float16 with `with torch.cuda.amp.autocast(enabled=False):`.  Does it mean for other opearation, you are using float16, but for the head prediction you are using float32?

**Comment by jytime** [comment]
Yes for all other operations, I used bfloat16 with torch autocast. For the head prediction, I used float32.

btw, have you git pulled the new code? I have added the code for gradient checkpoint there:

https://github.com/facebookresearch/vggt/blob/8e3d2809c1c93fa27298c290605841d7e876b285/vggt/models/aggregator.py#L275

**Comment by Zhanpeng1202** [comment]
Oh, thanks a lot for the information!!

Checking it right now. Have a nice rest of the day!
